lib/msal-browser/src/interaction_client/NativeInteractionClient.ts

```diff
@@ -236,6 +265,22 @@ export class NativeInteractionClient extends BaseInteractionClient {
 
         const instanceAware: boolean = !!(request.extraQueryParameters && request.extraQueryParameters.instance_aware);
 
+        // validate SHR parameters
+        if (request.authenticationScheme === AuthenticationScheme.POP) {
+            if (!request.resourceRequestMethod && !request.resourceRequestUri) {
```

Comment:
I thought these were technically optional?

Reply:
Or does wam shr require them?

Reply:
SHR always required them as far as I can check. Only `shrClaims` and `shrNonce` are optional.

Reply:
So yes, in future, when WAM will calculate the SHR, we need to mandate them. We never needed to before, as these never left the browser layer.

Reply:
This is from the docs, the code does not restrict anything as such. @hectormmg let's chat about this. I am also following up with WAM team on what they would like to mandate in the future. For now, my preference is to keep this and do an additional PR later with this data to proceed with a single branch testing for WAM JS Bridge.

Reply:
Reverting this, the decision is pending based on WAM response, for now, keeping the public surface intact.